src/domain_extended/data_structures.rs

```diff
@@ -0,0 +1,223 @@
+//! Data structures for the domain extended polynomial commitment scheme.
+//! Introduces
+//! * `DomainExtendedCommitment` and their `DomainExtendedRandomness` as vector variant
+//!    of the non-extended scheme, and implements group operations for these.
+//! * `DomainExtendedMultiPointProof` for the [HaloInfinite] batch evaluation proof of
+//!    the domain extended scheme, including implementations of serialization traits.
+//!
+//! [HaloInfinite]: https://eprint.iacr.org/2020/1536
+use crate::*;
+use algebra::groups::{Group, GroupVec};
+use std::{
+    io::{ Read, Write }, convert::TryFrom,
+};
+
+/// Multi-point multi-poly opening proof according to [BDFG2020](https://eprint.iacr.org/2020/081).
+/// Contains an extra (domain extended) commitment `h_comm` which cannot be reproduced by
+/// linear combinations.
+#[derive(Derivative)]
+#[derivative(
+Clone(bound = ""),
+Debug(bound = ""),
+Eq(bound = ""),
+PartialEq(bound = ""),
+)]
+pub struct DomainExtendedMultiPointProof<G, P>
+    where
+        G: Group,
+        P: PCProof,
+{
+
+    /// This is a "classical" single-point multi-poly proof which involves all commitments:
+    /// commitments from the initial claim and the new "h_commitment"
+    pub proof: P,
+
+    /// Commitment to the h(X) polynomial
+    pub h_commitment: GroupVec<G>,
```

Comment:
If you let GroupVec implement G: Curve, you could've reused the MultiPointProof struct defined in ipa_pc/data_structures.rs . Not sure if it's matematically sound though